src/olympia/addons/templates/addons/impala/details.html

```diff
@@ -145,14 +145,19 @@ <h3>{{ _('Enjoy this add-on?') }}</h3>
 
 {% if addon.type != amo.ADDON_PERSONA %}
   {% set version = addon.current_version %}
+  {% if addon.type == amo.ADDON_STATICTHEME %}
+    {% set previews = list(version.previews) %}
+  {% else %}
+    {% set previews = addon.all_previews %}
```

Comment:
Make `addon.all_previews` behave differently for static themes? That is, doing the `if` + use `addon.current_version.previews` thing you're doing here. Cause we'll want to return the correct previews for the API/frontend, right ?

There is a nice `FIXME` I added a while back in `Addon.attach_previews()` that plays in your favor, those themes have no `Preview`, so `attach_previews()` doesn't fill `all_previews`, so that still works... We might want to modify that transformer to be able to fetch `VersionPreview`s, too, though.

Reply:
I did consider overriding `Addon.all_previews` to return `VersionPreview` objects but I wasn't sure if we'd want _all_ previews for all versions to be returned (so it would actually be all previews) or just `current_version.previews`.  
The latter would mean less changes in code that consumes `Addon.all_previews`, I suppose, but could also mean we have to create a `really_all_previews` method for the times we _do_ want all of the previews that belong to an Addon.  (The values in ES for the Addon would need an update when the current_version changes too)
The former - returning a flat list for all versions - isn't particularly useful for end consumers expecting a list of Preview objects as it doesn't differentiate which Preview belongs to which Version - unless we make more changes to check for a VersionPreview in the right places.

We'd have to fake the missing properties (caption and addon) on the VersionPreview model too, though that's not difficult.

Reply:
I was thinking of just having `all_previews` use `current_version.previews`. Possibly renaming the property if necessary (can't be called `addon.previews`, sadly...)

Reply:
Well, it could, but we'd need to change the name of the `previews` backlink to be something else (`_previews`?  `raw_previews`? `addon_previews`?).

I'm not convinced that everywhere using `.previews` right now needs to, and shouldn't be using `.all_previews` instead, either, but I'll hold off from that refactoring.

Reply:
as mentioned on irc, this turns out to be not particularly useful in terms of avoiding rewriting code that uses `Addon.all_previews`.  Legacy API & discovery uses it but supporting static themes there isn't in scope (but would be easy to do).  API code uses `all_previews` too but requires some non-trivial changes to work around [ES]PreviewSerializer working on a Preview object specifically, and VersionPreviews.caption not being a TranslatedField

Reply:
Let's leave it for now then, but I'd really like to avoid having different places in the API results that return previews. It feels wrong to be pushing complexity to clients just because of internal implementation issues.

I'd like to kill `ESPreviewSerializer` btw: https://github.com/mozilla/addons-server/issues/7563#issuecomment-372465411

Reply:
#6804 deals with the API